src/Pact/Native.hs

```diff
@@ -1238,8 +1247,10 @@ stringToCharList :: Text -> V.Vector (Term a)
 stringToCharList t = V.fromList $ tLit . LString . T.singleton <$> T.unpack t
 
 strToList :: GasRNativeFun e
-strToList g i [TLitString s] = computeGas' g i (GMakeList $ fromIntegral $ T.length s) $
-  return $ toTListV tTyString def $ stringToCharList s
+strToList g i [TLitString s] = do
+  let !len = fromIntegral $ T.length s
```

Comment:
Here the bang does mean something else: Always compute the length, even if `ga` is never demanded. I don't think this is necessary, and would rather just let laziness call `T.length` when and if it needs to, since only one of the arguments to `ifExecutionFlagSet'` will ever be evaluated, so the bang pattern isn't saving time, it's just shifting the cost of the length from "maybe happens" to "always happens here".